### PR TITLE
[Messenger] Move Transport/InMemoryTransport to Transport/InMemory/InMemoryTransport

### PR DESCRIPTION
**Pull request by lyrixx**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       |
| License       | MIT
| Doc PR        |

---
For consistency with `SyncTransport`

### CODE REVIEW

**Comment by ktherage** [comment]
I don't know why carsonbot pinged me on that PR. 
Anyways I see no problems here.

**Review by OskarStark — APPROVED** [review]

File: UPGRADE-6.3.md [hunk 1]
```diff
@@ -11,6 +11,15 @@ HttpKernel
...
  * Deprecate parameters `container.dumper.inline_factories` and `container.dumper.inline_class_loader`, use `.container.dumper.inline_factories` and `.container.dumper.inline_class_loader` instead
 
+Messenger
+---------
+
+ * Deprecate `Symfony\Component\Messenger\Transport\InMemoryTransport` in favor
+   of `Symfony\Component\Messenger\Transport\InMemory\InMemoryTransport`
+ * Deprecate `Symfony\Component\Messenger\Transport\InMemoryTransportFactory` in
+   favor of
+   `Symfony\Component\Messenger\Transport\InMemory\InMemoryTransportFactory`
```
Comment (nicolas-grekas) [inline UPGRADE-6.3.md:20]: should be merged with the previous line

File: UPGRADE-7.0.md [hunk 2]
```diff
@@ -1,6 +1,15 @@
...
 =======================
 
+Messenger
+---------
+
+ * Remove `Symfony\Component\Messenger\Transport\InMemoryTransport` in favor
+   of `Symfony\Component\Messenger\Transport\InMemory\InMemoryTransport`
+ * Remove `Symfony\Component\Messenger\Transport\InMemoryTransportFactory` in
+   favor of
+   `Symfony\Component\Messenger\Transport\InMemory\InMemoryTransportFactory`
```
Comment (nicolas-grekas) [inline UPGRADE-7.0.md:10]: same

File: src/Symfony/Component/Messenger/CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,15 @@
...
 =========
 
+6.3
+---
+
+ * Deprecate `Symfony\Component\Messenger\Transport\InMemoryTransport` in favor
+   of `Symfony\Component\Messenger\Transport\InMemory\InMemoryTransport`
+ * Deprecate `Symfony\Component\Messenger\Transport\InMemoryTransportFactory` in
+   favor of
+   `Symfony\Component\Messenger\Transport\InMemory\InMemoryTransportFactory`
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/CHANGELOG.md:10]: here also

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.php [hunk 4]
```diff
@@ -35,7 +35,7 @@
 use Symfony\Component\Messenger\Middleware\ValidationMiddleware;
 use Symfony\Component\Messenger\Retry\MultiplierRetryStrategy;
 use Symfony\Component\Messenger\RoutableMessageBus;
-use Symfony\Component\Messenger\Transport\InMemoryTransportFactory;
+use Symfony\Component\Messenger\Transport\InMemory\InMemoryTransportFactory;
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/messenger.php:38]: composer.json should be bumped accordingly or we should handle both classes

**Review by nicolas-grekas — APPROVED** [review]
LGTM, just some minor things, thanks.

**Comment by lyrixx** [comment]
@nicolas-grekas Thanks for the review. I addressed your comments.

**Comment by nicolas-grekas** [comment]
Thank you @lyrixx.

**Pull request merged**